go-controller/pkg/ovn/address_set.go

```diff
@@ -418,3 +420,48 @@ func getIPv4ASHashedName(name string) string {
 func getIPv6ASHashedName(name string) string {
 	return hashedAddressSet(name + ipv6AddressSetSuffix)
 }
+
+// This method checks if an address set exists in OVN NB or not. In some cases when a namespace
```

Comment:
we could perhaps also add oc.waitForNamespaceLocked in pods.go add logical port when we add pod to namespace instead of getNamespaceLocked. It might prevent his from happening most of the time, with the con of potential add logical port delay of up to 10 seconds, @dcbw thoughts?

Reply:
my question about this race is, is the problem caused because of data inconsistency?
someone delete the address set and other is trying to use it?
or is because the handler that has to add the addressSet is lagging and this tries to use it before is created?
for the first one idea I think may work is to have a kind of repair loop verifying that ovn db and ovn-kube are in sync.
for the second one I'm afraid that locking will not solve it and will increase the risk of deadlock, how can we correctly streamline operations with handlers so they are executed sequentially in the right order? cc: @danwinship

Reply:
It's usually because of the second reason, asynchronous events. The thing is though, ovn-nbctl allows you to configure an ACL that references a non-existent address set. When OVN eventually sees the address set, it should just then create the right openflow flows. If there is a case where we never configure the address set, then that's a bug on our side.

For "I think may work is to have a kind of repair loop verifying that ovn db and ovn-kube are in sync." We are already doing some of that by caching things. The plan was to iterate on the work @dcbw did with gress_policy work and fully define a port group and cache the acls, address sets, etc that belong to that policy; I just haven't had time to do it yet.

Reply:
See:
https://github.com/ovn-org/ovn-kubernetes/pull/1068
https://github.com/ovn-org/ovn/issues/35#issuecomment-587128826

Reply:
Thanks @trozet for the link and explanation. GIven that information I think this fix may not be needed after all.